### PR TITLE
Docs: overview of the application logic

### PR DESCRIPTION
**Pull request by thekaveman**
Based on some recent discussion and onboarding new team members, this page provides a high-level overview of all of the main phases through which the user is taken by the app.

Preview page: https://benefits-2139--cal-itp-previews.netlify.app/development/application-logic/

### CODE REVIEW

File: docs/development/application-logic.md [hunk 1]
```diff
@@ -0,0 +1,287 @@
...
+Depending upon the choice of enrollment pathway, the _Next phase_ above may be:
+
+- [Identity proofing](#identity-proofing), for all flows that require user PII (such as for verifying age).
+- [Eligibility verification](#eligibility-verification), for Agency card flows that require a physical card from the transit
+  agency.
+
+## Identity proofing
+
+In this phase, Cal-ITP Benefits takes the user through an [OpenID Connect (OIDC)](https://openid.net/developers/how-connect-works/)
+flow as a Client (RP) of the CDT Identity Gateway (the Identity Provider or IDP), via Login.gov.
```
Comment (lalver1) [inline docs/development/application-logic.md:133]:
```suggestion
flow as a Client (the Relying Party or RP) of the CDT Identity Gateway (the Identity Provider or IDP), via Login.gov.
```

**Review by lalver1 — COMMENTED** [review]
This looks great! 👍  I think the `Entrypoint` and `Key supporting files` call-outs work well to link the descriptions in the documentation with the code. The last diagram in the [Enrollment](https://benefits-2139--cal-itp-previews.netlify.app/development/application-logic/#enrollment) section is also very useful given the different tokens used in this phase.
I only added one minor suggested change.

**Comment by thekaveman** [comment]
Thanks for the suggestion @lalver1, I added that in.

I also added to each of the diagrams a block for the interaction with analytics, it's helpful to see when the key events are generated in each phase.

**Comment by lalver1** [comment]
I agree, adding the interaction with analytics made the diagrams even more useful yet they still look clear.

**Review by lalver1 — APPROVED** [review]
Looks good 👍

**Review by angela-tran — APPROVED** [review]
This looks great! I really like the diagrams.

**Pull request merged**